File: experiment/switch_ifelse_dict/dictionary.js
function applyTuple(action, num) {
  var result;

  var action_handlers = {
    half: function() {
      result = num / 2;
    },
    double: function() {
      result = num * 2;
    },
    triple: function() {
      result = num * 3;
    },
    quadruple: function() {
      result = num * 4;
    },
    default: function() {
      result = num;
    }
  };

  (action_handlers[action] || action_handlers['default'])();
  
  return result;
}

const result = applyTuple('double', 2);

File: etc/dictionary.py
def applyTuple(action, num):
  action_handlers = {
    'half': lambda : num / 2,
    'double':  lambda : num * 2,
    'triple': lambda : num * 3,
    'quadruple': lambda : num * 4,
    'default': lambda : num
  }

  return (action_handlers.get(action) or action_handlers.get('default'))()

result = applyTuple('double', 2)
